pkg/operator/rules_test.go

```diff
@@ -198,9 +189,7 @@ func shouldResetRuleWithPartialResponseStrategySet(t *testing.T) {
 	promVersion, _ := semver.ParseTolerant(DefaultPrometheusVersion)
 	pr := newRuleSelectorForConfigGeneration(PrometheusFormat, promVersion)
 	content, _ := pr.generateRulesConfiguration(rules)
-	if strings.Contains(content, "partial_response_strategy") {
-		t.Fatalf("expected `partial_response_strategy` removed from PrometheusRule")
-	}
+	require.False(t, strings.Contains(content, "partial_response_strategy"), "expected `partial_response_strategy` removed from PrometheusRule")
```

Comment:
here you could use https://pkg.go.dev/github.com/stretchr/testify/require#NotContains